frontend/src/components/auth/Settings.tsx

```diff
@@ -0,0 +1,758 @@
+                placeholder={PLACEHOLDER_MOBILE_EXAMPLE_FULL_NAME}
+                onChange={(e) => {
+                  onsiteInfo[index].name = e.target.value;
+                  setOnsiteInfo([...onsiteInfo]);
+                }}
+              />
+            </FormControl>
+            <FormControl
+              isRequired={index === 0}
+              isInvalid={attemptedSubmit && onsiteInfo[index].phone === ""}
+            >
+              <Input
+                h="37px"
+                variant="mobile-outline"
+                type="tel"
+                value={onsiteInfo[index].phone}
+                placeholder={PLACEHOLDER_MOBILE_EXAMPLE_PHONE_NUMBER}
+                onChange={(e) => {
+                  onsiteInfo[index].phone = e.target.value;
+                  setOnsiteInfo([...onsiteInfo]);
+                }}
+              />
+            </FormControl>
+            <FormControl
+              isRequired={index === 0}
+              isInvalid={
+                attemptedSubmit && !isValidEmail(onsiteInfo[index].email)
+              }
+            >
+              <Input
+                h="37px"
+                variant="mobile-outline"
+                type="email"
+                value={onsiteInfo[index].email}
+                placeholder={PLACEHOLDER_MOBILE_EXAMPLE_EMAIL}
+                onChange={(e) => {
+                  onsiteInfo[index].email = e.target.value;
+                  setOnsiteInfo([...onsiteInfo]);
+                }}
+              />
+            </FormControl>
+          </Flex>
+        ))}
+        {onsiteInfo.length < 10 && (
+          <Text
+            variant="mobile-body-bold"
+            color="primary.blue"
+            cursor="pointer"
+            w="fit-content"
+            onClick={() => {
+              setOnsiteInfo([
+                ...onsiteInfo,
+                {
+                  name: "",
+                  phone: "",
+                  email: "",
+                },
+              ]);
+            }}
+          >
+            + Add another contact
+          </Text>
+        )}
+      </Flex>
+    );
+  };
+
+  const isRequestValid = (): boolean => {
+    const stringsToValidate = [
+      primaryContact.name,
+      organizationName,
+      organizationAddress,
+    ];
+    const phoneNumsToValidate = [primaryContact.phone];
+    const emailsToValidate = [primaryContact.email];
+
+    for (let i = 0; i < onsiteInfo.length; i += 1) {
+      stringsToValidate.push(onsiteInfo[i].name);
+      phoneNumsToValidate.push(onsiteInfo[i].phone);
+      emailsToValidate.push(onsiteInfo[i].email);
+    }
+
+    for (let i = 0; i < stringsToValidate.length; i += 1) {
+      if (stringsToValidate[i] === "") return false;
+    }
+
+    for (let i = 0; i < phoneNumsToValidate.length; i += 1) {
+      if (phoneNumsToValidate[i] === "") return false;
+    }
+
+    for (let i = 0; i < emailsToValidate.length; i += 1) {
+      if (!isValidEmail(emailsToValidate[i])) return false;
```

Comment:
BTW I think that just relying on the html email validation is probably good enough for the FE: https://www.w3schools.com/tags/att_input_type_email.asp

No action needed, more validation is better! But just wanted to mention it.